projects/vdk-plugins/vdk-structlog/src/vdk/plugin/structlog/structlog_plugin.py

```diff
@@ -77,14 +81,152 @@ def process(self, msg, kwargs):
 """
 
 
+class StructlogConfig:
+    def __init__(self, configuration: Configuration):
+        presets = {
+            "LOCAL": {
+                STRUCTLOG_USE_STRUCTLOG: configuration.get_value(
+                    STRUCTLOG_USE_STRUCTLOG
+                ),
+                STRUCTLOG_LOGGING_METADATA_KEY: ",".join(
+                    list(JSON_STRUCTLOG_LOGGING_METADATA_DEFAULT.keys())
+                ),
+                STRUCTLOG_LOGGING_FORMAT_KEY: STRUCTLOG_LOGGING_FORMAT_DEFAULT,
+                STRUCTLOG_CONSOLE_LOG_PATTERN: "",
+                STRUCTLOG_CONFIG_PRESET: configuration.get_value(
+                    STRUCTLOG_CONFIG_PRESET
+                ),
+                SYSLOG_HOST_KEY: DEFAULT_SYSLOG_HOST,
+                SYSLOG_PORT_KEY: DEFAULT_SYSLOG_PORT,
+                SYSLOG_PROTOCOL_KEY: DEFAULT_SYSLOG_PROTOCOL,
+                SYSLOG_ENABLED_KEY: DEFAULT_SYSLOG_ENABLED,
+                vdk_config.LOG_LEVEL_VDK.lower(): configuration.get_value(
+                    vdk_config.LOG_LEVEL_VDK.lower()
+                ),
+                vdk_config.LOG_LEVEL_MODULE.lower(): configuration.get_value(
+                    vdk_config.LOG_LEVEL_MODULE.lower()
+                ),
+            },
+            "CLOUD": {
+                STRUCTLOG_USE_STRUCTLOG: configuration.get_value(
+                    STRUCTLOG_USE_STRUCTLOG
+                ),
+                STRUCTLOG_LOGGING_METADATA_KEY: "",
+                STRUCTLOG_LOGGING_FORMAT_KEY: STRUCTLOG_LOGGING_FORMAT_DEFAULT,
+                STRUCTLOG_CONSOLE_LOG_PATTERN: DETAILED_LOGGING_FORMAT,
+                STRUCTLOG_CONFIG_PRESET: configuration.get_value(
+                    STRUCTLOG_CONFIG_PRESET
+                ),
+                SYSLOG_HOST_KEY: DEFAULT_SYSLOG_HOST,
+                SYSLOG_PORT_KEY: DEFAULT_SYSLOG_PORT,
+                SYSLOG_PROTOCOL_KEY: DEFAULT_SYSLOG_PROTOCOL,
+                SYSLOG_ENABLED_KEY: DEFAULT_SYSLOG_ENABLED,
+                vdk_config.LOG_LEVEL_VDK.lower(): configuration.get_value(
+                    vdk_config.LOG_LEVEL_VDK.lower()
+                ),
+                vdk_config.LOG_LEVEL_MODULE.lower(): configuration.get_value(
+                    vdk_config.LOG_LEVEL_MODULE.lower()
+                ),
+            },
+        }
+
+        self._config = presets[configuration.get_value(STRUCTLOG_CONFIG_PRESET)]
+
+        for key in configuration.list_config_keys():
+            value = configuration.get_value(key)
```

Comment:
Are you sure the capital case would match? You may consider user key.lower if unsure

Reply:
This is ok, because all keys in the configuration get lower-cased for some reason.